### PR TITLE
Fix bug preventing multiple browsers from getting notified of changes

### PR DESCRIPTION
**Pull request by joeosburn**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -112,13 +112,16 @@ exports.balm = (opts = {}) => {
     _changes = [...new Set(_changes)];
 
     _timer = setTimeout(() => {
+      if (!_changes.length) return;
```
Comment (shanebo) [inline lib/index.js:115]: I'm wondering why I even had a _changes.length check if notifyClient only fires when something changed. Can you think through whether we even need it?

Comment (joeosburn) [reply lib/index.js:115]: maybe if there is a scenario if the timeout is running and another timeout gets registered while the `.send()` is happening? If that is even possible. That's only thing I can think of.

File: lib/config.js [hunk 2]
```diff
@@ -6,7 +6,17 @@ const { dirname, basename, resolve, extname } = require('path');
...
+
+const numberFromString = (str) => {
+  let h = 0, l = str.length, i = 0;
+
+  if (l > 0)
+    while (i < l)
+      h = (h << 5) - h + str.charCodeAt(i++) | 0;
+
+  return parseInt(h.toString().substring(1, 5));
+};
```
Comment (shanebo) [inline lib/config.js:19]: I would update to this:

```
  const toPort = (path) => {
    let i = 0;
    let h = 0;
    let l = path.length;

    if (l > 0) {
      while (i < l) {
        h = (h << 5) - h + path.charCodeAt(i++) | 0;
      }
    }

    return 3000 + parseInt(h.toString().substring(1, 5));
  }
```

File: lib/config.js [hunk 3]
```diff
@@ -92,7 +102,7 @@ module.exports = (opts = {}) => {
     templates: {},
     handles: {},
     loadHandles: true,
-    port: makeRandomPort(),
+    port: generatePortFromPath(),
```
Comment (shanebo) [inline lib/config.js:105]: `port: toPort(__dirname),`

File: lib/hmr.js [hunk 4]
```diff
@@ -16,6 +16,8 @@ function handleChange(file) {
   }
 }
 
-socket.onmessage = (e) => {
-  JSON.parse(e.data).forEach(handleChange);
-}
+const socket = new Sockette('ws://127.0.0.1:PORT', {
+  onmessage: e => {
```
Comment (shanebo) [inline lib/hmr.js:20]: `onmessage: (e) => {`

File: package.json [hunk 5]
```diff
@@ -23,6 +23,7 @@
     "klona": "~2.0.5",
     "merge-anything": "~4.0.2",
     "normalize-selector": "0.2.0",
+    "sockette": "^2.0.6",
```
Comment (shanebo) [inline package.json:26]: Should this be an `^` or a `~`?

**Review by shanebo — CHANGES_REQUESTED** [review]

File: lib/config.js (hunk 2 above)
Comment (joeosburn) [reply lib/config.js:19]: thanks

**Review by shanebo — APPROVED** [review]

**Pull request merged**